builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -74,6 +74,26 @@ func resourceAwsDbInstance() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !regexp.MustCompile(`^[0-9a-z-]$`).MatchString(value) {
```

Comment:
Should this be `[0-9a-z-]$` instead of `^[0-9a-z-]$` ?

`foobarbaz-test-terraform-test2` is failing:

http://rubular.com/r/ixQHBNXPiV vs http://rubular.com/r/difJw4AHVJ